### PR TITLE
Spatial submission ian

### PR DESCRIPTION
**Pull request by ianliuwk1019**
This PR contains codes for 'processSpatialSubmission' related logic. This can now partially save spatial objects into cut_block/road_section/WTRA from the Swagger OpenAPI page.
- Add 'cascade' relation for submission and spatial object entities
- DTO (submission-with-json.dto.ts) Extends from TypeORM's 'OmitType' temporary so the generated OpenAPI has the correct properties when POST.
- Adjust service method logic to save/replace spatial objects for the submission.

### CODE REVIEW

File: apps/api/src/app/controllers/cut-block/entities/cut-block.entity.ts [hunk 1]
```diff
@@ -30,11 +32,15 @@ export class CutBlock extends ApiBaseEntity<CutBlock> {
...
+  @ManyToOne(() => Submission, (submission) => submission.cut_blocks, {onDelete: 'CASCADE', orphanedRowAction:'delete'})
   @JoinColumn({ name: 'submission_id', referencedColumnName: 'id' })
   submission: Submission;
 
   @Column()
   @RelationId((cutBlock: CutBlock) => cutBlock.submission)
   submission_id: number;
+
+  @OneToMany(type => PublicComment, (publicComment) => publicComment.cutBlock, 
+    {cascade: true})
```
Comment (basilv) [inline apps/api/src/app/controllers/cut-block/entities/cut-block.entity.ts:44]: We don't want / need a relationship to PublicComment from CutBlock.

File: apps/api/src/app/controllers/public-comment/entities/public-comment.entity.ts [hunk 2]
```diff
@@ -65,6 +66,14 @@ export class PublicComment extends ApiBaseEntity<PublicComment> {
   @Column()
   scope_cut_block_id: number;
 
+  @ManyToOne(() => CutBlock, (cutBlock) => cutBlock.publicComments, {onDelete: 'CASCADE', orphanedRowAction:'delete'})
+  @JoinColumn({ name: 'cutBlock_id', referencedColumnName: 'id' })
+  cutBlock: CutBlock;
+
+  @Column({name: "scope_cut_block_id"})
+  @RelationId((publicComment: PublicComment) => publicComment.cutBlock)
```
Comment (basilv) [inline apps/api/src/app/controllers/public-comment/entities/public-comment.entity.ts:74]: We don't need a relationship from PublicComment to cutblock.

File: apps/api/src/app/controllers/submission/dto/submission-with-json.dto.ts [hunk 3]
```diff
@@ -16,11 +17,16 @@ export enum SpatialObjectCodeEnum {
   WTRA = 'WTRA'
 };
 
-export class SubmissionWithJsonDto extends SubmissionDto {
+// export class SubmissionWithJsonDto extends SubmissionDto {
+export class SubmissionWithJsonDto extends OmitType(SubmissionDto, 
+  // TODO: perhaps, it is better to create a new "createDTO" with below annotations there and extends from that DTO.
```
Comment (basilv) [inline apps/api/src/app/controllers/submission/dto/submission-with-json.dto.ts:22]: We might be just better off creating a completely separate Dto not inheriting from a base class.

Comment (ianliuwk1019) [reply apps/api/src/app/controllers/submission/dto/submission-with-json.dto.ts]: Thanks. I removed the 'extends' with parent class making it its own class.
I was thinking to rename it but then didn't change and still keep the class name, hope it is ok.

**Pull request merged**